test/list.go

```diff
@@ -335,22 +332,17 @@ func DescribeListTestsFor(ctx *common.TestContext, t TestCase) bool {
 
 		By("[TEST]: ====================================")
 
-		req := ctx.SMWithOAuth.GET(t.API)
-		if query != "" {
-			req = req.WithQueryString(query)
-		}
-
 		By(fmt.Sprintf("[TEST]: Verifying expected status code %d is returned from list operation", listOpEntry.expectedStatusCode))
-		resp := req.
-			Expect().
-			Status(listOpEntry.expectedStatusCode)
 
 		if listOpEntry.expectedStatusCode != http.StatusOK {
 			By(fmt.Sprintf("[TEST]: Verifying error and description fields are returned after list operation"))
-
-			resp.JSON().Object().Keys().Contains("error", "description")
+			req := ctx.SMWithOAuth.GET(t.API)
```

Comment:
this should probably be List and not Get and should happen before the status code branching, imo. Normally the method that the test tests is independant from the expectations (calling GET if you expect error and list otherwise seems a bit confusing

Reply:
The List method expects the iteration to be successful and returns the array of items.
Here we want to assert the error is in the response which is different case and no need to iterate or make multiple calls. If this checks are moved to List and the status code expected is moved to the List it will not look like List anymore.